### PR TITLE
Add bit vector support for 1024

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds bit vector support for length at 1024.

This is needed to support the upcoming light client object for beacon chain

### CODE REVIEW

File: bitvector1024.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	"math/bits"
+)
+
+var _ = Bitfield(Bitvector1024{})
+
+// Bitvector1024 is a bitfield with a fixed defined size of 1024. There is no length bit
+// present in the underlying byte array.
+type Bitvector1024 []byte
+
+// NewBitvector1024 creates a new bitvector of size 64.
```
Comment (rkapka) [inline bitvector1024.go:14]:
```suggestion
// NewBitvector1024 creates a new bitvector of size 1024.
```

File: bitvector1024.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+	bit := uint8(1 << (idx % 8))
+	if val {
+		b[idx/8] |= bit
+	} else {
+		b[idx/8] &^= bit
+	}
+}
+
+// Len returns a constant length 64.
```
Comment (rkapka) [inline bitvector1024.go:49]: What does `constant length 64` mean?

File: bitvector1024.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		return 0
+	}
+	c := 0
+	for _, bt := range b {
+		c += bits.OnesCount8(bt)
+	}
+	return uint64(c)
+}
+
+// Bytes returns the bytes data representing the bitvector64. This method
```
Comment (rkapka) [inline bitvector1024.go:66]:
```suggestion
// Bytes returns the bytes data representing the bitvector1024. This method
```

File: bitvector1024.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		num <<= uint8(i)
+		binary.BigEndian.PutUint64(b, num)
+	} else {
+		num := binary.BigEndian.Uint64(b)
+		num >>= uint8(i * -1)
+		binary.BigEndian.PutUint64(b, num)
+	}
+}
+
+func (b Bitvector1024) BitIndices() []int {
```
Comment (rkapka) [inline bitvector1024.go:100]: This function has no comment.

File: bitvector1024_test.go [hunk 5]
```diff
@@ -0,0 +1,463 @@
...
+import (
+	"bytes"
+	"reflect"
+	"testing"
+)
+
+func TestBitvector1024_Len(t *testing.T) {
+	bv := NewBitvector1024()
+	if bv.Len() != 1024 {
+		t.Errorf("(%x).Len() = %d, wanted %d", bv, bv.Len(), 64)
```
Comment (rkapka) [inline bitvector1024_test.go:12]:
```suggestion
		t.Errorf("(%x).Len() = %d, wanted %d", bv, bv.Len(), 1024)
```

File: bitvector1024_test.go [hunk 6]
```diff
@@ -0,0 +1,463 @@
...
+	}
+}
+
+func TestBitvector1024_BitAt(t *testing.T) {
```
Comment (rkapka) [inline bitvector1024_test.go:16]: https://github.com/prysmaticlabs/go-bitfield/blob/master/bitvector64_test.go includes an out of bounds case. Is there such a case here? If yes, can you please tag it with a comment?

Comment (terencechain) [reply bitvector1024_test.go:16]: It has one. I added a comment for clarification

https://github.com/prysmaticlabs/go-bitfield/pull/33/commits/47ed996352ff1b924c11b9953fe8b51c5b09afb5#diff-28964f5e352b039b26d03a3a9d92fe63423cee94d9f8d341c63926bb6531d10dR92

**Review by rkapka — APPROVED** [review]

**Pull request merged**